### PR TITLE
Make it easier to click on addons in the Addons list

### PR DESCRIPTION
**Pull request by brandonsturgeon**
### Summary
While managing lots of addons, it's really nice to have the checkbox for bulk actions.

However, it's also kind of cumbersome to _snipe_ the checkbox on each Addon when you're trying to select a lot of them at once.

This PR lets players click anywhere within the Addon panel to select/deselect the Addon.

### Demo
https://github.com/Facepunch/garrysmod/assets/7936439/987161e8-66ca-494f-95a6-1680ea74563c

### Details

The PR is actually really simple. A new click handler is added to the `workshopicon` element that will Toggle the Selected state of the Addon.

https://github.com/Facepunch/garrysmod/assets/7936439/8c10ca9d-28d5-4c12-b4b0-14cb4500b9cc

### CODE REVIEW

**Review by robotboy655 — CHANGES_REQUESTED** [review]
Actually, this works even better:

```lua
	$scope.ToggleSelect = function( file, event )
	{
		var en = event.target.nodeName.toLowerCase();
		if ( en != "controls" && en != "description" && en != "workshopicon" ) return;

		$scope.SelectedItems[file.id] = !$scope.SelectedItems[file.id];
		if ( event ) event.stopPropagation();
	}
```

**Comment by brandonsturgeon** [comment]
> I like the idea (despite the showcase)

Sorry, it was all I was subscribed to at the time.


> Actually, this works even better:
> 
> ```js
> 	$scope.ToggleSelect = function( file, event )
> 	{
> 		var en = event.target.nodeName.toLowerCase();
> 		if ( en != "controls" && en != "description" && en != "workshopicon" ) return;
> 
> 		$scope.SelectedItems[file.id] = !$scope.SelectedItems[file.id];
> 		if ( event ) event.stopPropagation();
> 	}
> ```

I'll give this a try 👍

**Comment by brandonsturgeon** [comment]
That was a great suggestion - works great!

Thank you - those changes were applied.

**Review by robotboy655 — APPROVED** [review]

File: garrysmod/html/template/addon_list.html [hunk 1]
```diff
@@ -68,7 +68,7 @@ <h1 class="menuheader">
 				<size ng-hide="!file.info.size">{{GetNiceSize( file.info.size )}}</size>
 				<votes ng-show="!file.local && (file.info.up-file.info.down) > 0" >+{{file.info.up-file.info.down|number:0}}</votes>
 				<votes ng-show="!file.local && (file.info.up-file.info.down) < 0" class="negative">{{file.info.up-file.info.down|number:0}}</votes>
-				<description>{{GetAddonDescription( file )}}</description>
+				<description ng-click="ToggleSelect( file, $event )">{{GetAddonDescription( file )}}</description>
```
Comment (robotboy655) [inline garrysmod/html/template/addon_list.html:71]: ng-click can be applied to `workshopicon` with the changes to `ToggleSelect`

**Review by robotboy655 — CHANGES_REQUESTED** [review]

**Review by robotboy655 — APPROVED** [review]

**Pull request merged**